src/Pages/Products/ProductDetails/SlideCarousel.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from "react";
+import styled, { css } from "styled-components";
+import Slide from "./Slide";
+import Thumbnail from "./Thumbnail";
+
+const SlideCarousel = () => {
+  const [optionData, setOptionData] = useState([]);
+
+  useEffect(() => {
+    fetch("/data/data.json")
+      .then((res) => res.json())
+      .then((res) => setOptionData(res.option));
+  }, []);
+
+  const [currentSlide, setCurrentSlide] = useState(0);
+  const [positionX, setPositionX] = useState(0);
+
+  const nextSlide = () => {
+    let TOTAL_SLIDES = optionData.length;
+    if (currentSlide >= TOTAL_SLIDES) {
+      return;
+    } else {
+      setCurrentSlide(currentSlide + 1);
+      setPositionX(positionX - 700);
+    }
```

Comment:
currentSlide가 TOTAL_SLIDES 보다 작을 때만 해당 set 함수들을 불러주면 되겠네요.
```suggestion
    if (currentSlide < TOTAL_SLIDES) {
      setCurrentSlide(currentSlide + 1);
      setPositionX(positionX - 700);
    }
```